### PR TITLE
Default python logging

### PR DESCRIPTION
**Pull request by JeffreyDevloo**

### CODE REVIEW

File: ovs/lib/tests/generic_tests/test_generic.py [hunk 1]
```diff
@@ -18,25 +18,26 @@
...
 from ovs.dal.lists.servicetypelist import ServiceTypeList
 from ovs.dal.tests.helpers import DalHelper
 from ovs.extensions.db.arakooninstaller import ArakoonClusterConfig, ArakoonInstaller
 from ovs.extensions.generic.configuration import Configuration
-from ovs_extensions.log.logger import Logger
 from ovs.extensions.generic.sshclient import SSHClient, UnableToConnectException
 from ovs_extensions.generic.tests.sshclient_mock import MockedSSHClient
 from ovs.lib.generic import GenericController
 from ovs.lib.helpers.toolbox import Toolbox
+from ovs_extensions.testing.testcase import LogTestCase
```
Comment (sjovdnbos) [inline ovs/lib/tests/generic_tests/test_generic.py:33]: alphabetical

File: ovs/cli/commands.py [hunk 2]
```diff
@@ -13,23 +13,38 @@
...
+
+    def invoke(self, ctx):
         """
-        Setup and add nullHandler by default
+        Invoke the command
         """
-        super(LogTestCase, self).setUp()
-        root_logger = logging.getLogger()
-        root_logger.addHandler(logging.NullHandler())
+        # from ovs.extensions.log import configure_logging
```
Comment (sjovdnbos) [inline ovs/cli/commands.py]: Intended?

File: ovs/celery_run.py [hunk 3]
```diff
@@ -32,9 +34,11 @@
 from kombu import Queue
 from kombu.serialization import register
 from threading import Thread
+from ovs_extensions.constants import is_unittest_mode
+from ovs.constants.logging import CELERY_RUN_LOGGER, OVS_LOGGER
```
Comment (sjovdnbos) [inline ovs/celery_run.py:38]: Imports alphabetical please

File: ovs/constants/logging.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# as published by the Free Software Foundation, in version 3 as it comes
+# in the LICENSE.txt file of the Open vStorage OSE distribution.
+#
+# Open vStorage is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY of any kind.
+
+OVS_LOGGER = 'ovs'
+API_LOGGER = 'api'
+
+UPDATE_LOGGER = '{}.update'.format(OVS_LOGGER)
```
Comment (sjovdnbos) [inline ovs/constants/logging.py]: We conventionally use `{0}`. I see no reason to not do that here

File: ovs/extensions/generic/heartbeat.py [hunk 5]
```diff
@@ -31,7 +32,7 @@ class HeartBeat(object):
     """
     ARP_TIMEOUT = 30
 
-    logger = Logger('extensions-generic')
+    logger = logging.getLogger('ovs.heartbeat')
```
Comment (sjovdnbos) [inline ovs/extensions/generic/heartbeat.py:35]: could be offloaded to constants too

**Review by sjovdnbos — CHANGES_REQUESTED** [review]

File: ovs/extensions/log.py [hunk 6]
```diff
@@ -38,66 +39,277 @@
...
-                          'ovs': OVS_FORMATTER_CONFIG
-                      },
-                      'handlers': DEFAULT_LOG_HANDLER_CONFIG,
-                      'loggers': {OVS_LOGGER: DEFAULT_LOGGER_CONFIG,
-                                  EXTENSIONS_LOGGER_NAME: DEFAULT_LOGGER_CONFIG,
-                                  API_LOGGER: DEFAULT_LOGGER_CONFIG,
-                                  'urllib3': {'level': 'WARNING'},
-                                  'paramiko': {'level': 'WARNING'},
-                                  'requests': {'level': 'WARNING'}}}
+LIBRARY_LOG_CONFIGS = {'urllib3': {'level': 'WARNING'},
```
Comment (sjovdnbos) [inline ovs/extensions/log.py:49]: Is it necessary to put this into a variable?

File: ovs/extensions/log.py [hunk 7]
```diff
@@ -38,66 +39,277 @@
...
 def get_logging_info():
+    # type: () -> LogInfoContainer
     """
     Retrieve logging information from the Configuration management
     :return: Dictionary containing logging information
     :rtype: dict
     """
     try:
-        return Configuration.get('/ovs/framework/logging')
+        return LogInfoContainer(**Configuration.get('/ovs/framework/logging'))
```
Comment (sjovdnbos) [inline ovs/extensions/log.py]: `/ovs/framework/logging`: offload to fwk-extensions: constants-framework

**Review by sjovdnbos — CHANGES_REQUESTED** [review]

File: ovs/extensions/log.py (hunk 6 above)
Comment (JeffreyDevloo) [reply ovs/extensions/log.py:49]: Just to separate the different loggers

**Review by sjovdnbos — APPROVED** [review]

**Pull request merged**